From d803dd359dd431cad08e95952f90c3f4f5acc9b7 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Sat, 12 Jan 2013 13:30:37 +0000
Subject: [PATCH 095/364] Clean up dangling references to grub-setup. Fixes
 Ubuntu bug #1082045.

* docs/grub.texi (Images): Refer generally to grub-install rather
than directly to grub-setup.
(Installing GRUB using grub-install): Remove direct reference to
grub-setup.
(Device map) Likewise.
(Invoking grub-install): Likewise.
* docs/man/grub-install.h2m (SEE ALSO): Likewise.
* docs/man/grub-mkimage.h2m (SEE ALSO): Likewise.
* util/grub-install.in (usage): Likewise.

* util/bash-completion.d/grub-completion.bash.in (_grub_setup):
Apply to grub-bios-setup and grub-sparc64-setup rather than to
grub-setup.
* configure.ac: Remove grub_setup output variable.

* docs/man/grub-bios-setup.h2m (NAME): Change name from grub-setup
to grub-bios-setup.
* docs/man/grub-sparc64-setup.h2m (NAME): Change name from
grub-setup to grub-sparc64-setup.
---
 ChangeLog                                      | 25 +++++++++++++++++++++++++
 configure.ac                                   |  1 -
 docs/grub.texi                                 | 24 ++++++++++++------------
 docs/man/grub-bios-setup.h2m                   |  2 +-
 docs/man/grub-install.h2m                      |  1 -
 docs/man/grub-mkimage.h2m                      |  1 -
 docs/man/grub-sparc64-setup.h2m                |  2 +-
 util/bash-completion.d/grub-completion.bash.in | 14 ++++++++++----
 util/grub-install.in                           |  4 ++--
 9 files changed, 51 insertions(+), 23 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 22b18b1..61bf8e7 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,28 @@
+2013-01-12  Colin Watson  <cjwatson@ubuntu.com>
+
+	Clean up dangling references to grub-setup.
+	Fixes Ubuntu bug #1082045.
+
+	* docs/grub.texi (Images): Refer generally to grub-install rather
+	than directly to grub-setup.
+	(Installing GRUB using grub-install): Remove direct reference to
+	grub-setup.
+	(Device map) Likewise.
+	(Invoking grub-install): Likewise.
+	* docs/man/grub-install.h2m (SEE ALSO): Likewise.
+	* docs/man/grub-mkimage.h2m (SEE ALSO): Likewise.
+	* util/grub-install.in (usage): Likewise.
+
+	* util/bash-completion.d/grub-completion.bash.in (_grub_setup):
+	Apply to grub-bios-setup and grub-sparc64-setup rather than to
+	grub-setup.
+	* configure.ac: Remove grub_setup output variable.
+
+	* docs/man/grub-bios-setup.h2m (NAME): Change name from grub-setup
+	to grub-bios-setup.
+	* docs/man/grub-sparc64-setup.h2m (NAME): Change name from
+	grub-setup to grub-sparc64-setup.
+
 2013-01-11  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	Import gcrypt public-key cryptography and implement signature checking.
diff --git a/configure.ac b/configure.ac
index dde954e..92b550a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,7 +64,6 @@ grub_TRANSFORM([grub-probe])
 grub_TRANSFORM([grub-reboot])
 grub_TRANSFORM([grub-script-check])
 grub_TRANSFORM([grub-set-default])
-grub_TRANSFORM([grub-setup])
 grub_TRANSFORM([grub-sparc64-setup])
 
 # Optimization flag.  Allow user to override.
diff --git a/docs/grub.texi b/docs/grub.texi
index a92bd96..e75bae9 100644
--- a/docs/grub.texi
+++ b/docs/grub.texi
@@ -622,11 +622,11 @@ This install doesn't conflict with standard install as long as they are in
 separate directories.
 
 Note that @command{grub-install} is actually just a shell script and the
-real task is done by @command{grub-mkimage} and @command{grub-setup}.
-Therefore, you may run those commands directly to install GRUB, without
-using @command{grub-install}. Don't do that, however, unless you are very
-familiar with the internals of GRUB. Installing a boot loader on a running
-OS may be extremely dangerous.
+real task is done by other tools such as @command{grub-mkimage}. Therefore,
+you may run those commands directly to install GRUB, without using
+@command{grub-install}. Don't do that, however, unless you are very familiar
+with the internals of GRUB. Installing a boot loader on a running OS may be
+extremely dangerous.
 
 @node Making a GRUB bootable CD-ROM
 @section Making a GRUB bootable CD-ROM
@@ -688,8 +688,8 @@ storage devices.
 @section The map between BIOS drives and OS devices
 
 If the device map file exists, the GRUB utilities (@command{grub-probe},
-@command{grub-setup}, etc.) read it to map BIOS drives to OS devices.  This
-file consists of lines like this:
+etc.) read it to map BIOS drives to OS devices.  This file consists of lines
+like this:
 
 @example
 (@var{device}) @var{file}
@@ -2283,8 +2283,8 @@ bytes.
 The sole function of @file{boot.img} is to read the first sector of the core
 image from a local disk and jump to it.  Because of the size restriction,
 @file{boot.img} cannot understand any file system structure, so
-@command{grub-setup} hardcodes the location of the first sector of the core
-image into @file{boot.img} when installing GRUB.
+@command{grub-install} hardcodes the location of the first sector of the
+core image into @file{boot.img} when installing GRUB.
 
 @item diskboot.img
 This image is used as the first sector of the core image when booting from a
@@ -4689,9 +4689,9 @@ GRUB.
 @node Invoking grub-install
 @chapter Invoking grub-install
 
-The program @command{grub-install} installs GRUB on your drive using
-@command{grub-mkimage} and (on some platforms) @command{grub-setup}.  You
-must specify the device name on which you want to install GRUB, like this:
+The program @command{grub-install} generates a GRUB core image using
+@command{grub-mkimage} and installs it on your system.  You must specify the
+device name on which you want to install GRUB, like this:
 
 @example
 grub-install @var{install_device}
diff --git a/docs/man/grub-bios-setup.h2m b/docs/man/grub-bios-setup.h2m
index eebe3ef..ac6ede3 100644
--- a/docs/man/grub-bios-setup.h2m
+++ b/docs/man/grub-bios-setup.h2m
@@ -1,5 +1,5 @@
 [NAME]
-grub-setup \- set up a device to boot using GRUB
+grub-bios-setup \- set up a device to boot using GRUB
 [SEE ALSO]
 .BR grub-install (8),
 .BR grub-mkimage (1),
diff --git a/docs/man/grub-install.h2m b/docs/man/grub-install.h2m
index 2de371a..8cbbc87 100644
--- a/docs/man/grub-install.h2m
+++ b/docs/man/grub-install.h2m
@@ -3,5 +3,4 @@ grub-install \- install GRUB to a device
 [SEE ALSO]
 .BR grub-mkconfig (8),
 .BR grub-mkimage (1),
-.BR grub-setup (8),
 .BR grub-mkrescue (1)
diff --git a/docs/man/grub-mkimage.h2m b/docs/man/grub-mkimage.h2m
index ca08b0c..f0fbc2b 100644
--- a/docs/man/grub-mkimage.h2m
+++ b/docs/man/grub-mkimage.h2m
@@ -2,6 +2,5 @@
 grub-mkimage \- make a bootable image of GRUB
 [SEE ALSO]
 .BR grub-install (8),
-.BR grub-setup (8),
 .BR grub-mkrescue (1),
 .BR grub-mknetdir (8)
diff --git a/docs/man/grub-sparc64-setup.h2m b/docs/man/grub-sparc64-setup.h2m
index eebe3ef..18f803a 100644
--- a/docs/man/grub-sparc64-setup.h2m
+++ b/docs/man/grub-sparc64-setup.h2m
@@ -1,5 +1,5 @@
 [NAME]
-grub-setup \- set up a device to boot using GRUB
+grub-sparc64-setup \- set up a device to boot using GRUB
 [SEE ALSO]
 .BR grub-install (8),
 .BR grub-mkimage (1),
diff --git a/util/bash-completion.d/grub-completion.bash.in b/util/bash-completion.d/grub-completion.bash.in
index 5f4b249..44bf135 100644
--- a/util/bash-completion.d/grub-completion.bash.in
+++ b/util/bash-completion.d/grub-completion.bash.in
@@ -252,10 +252,16 @@ _grub_setup () {
         _filedir
     fi
 }
-__grub_setup_program="@grub_setup@"
-have ${__grub_setup_program} && \
- complete -F _grub_setup -o filenames ${__grub_setup_program}
-unset __grub_setup_program
+
+__grub_bios_setup_program="@grub_bios_setup@"
+have ${__grub_bios_setup_program} && \
+ complete -F _grub_setup -o filenames ${__grub_bios_setup_program}
+unset __grub_bios_setup_program
+
+__grub_sparc64_setup_program="@grub_sparc64_setup@"
+have ${__grub_sparc64_setup_program} && \
+ complete -F _grub_setup -o filenames ${__grub_sparc64_setup_program}
+unset __grub_sparc64_setup_program
 
 
 #
diff --git a/util/grub-install.in b/util/grub-install.in
index 218bbd9..aac27c7 100644
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -114,8 +114,8 @@ echo
 gettext "INSTALL_DEVICE must be system device filename.";echo
 echo
 
-gettext_printf "%s copies GRUB images into %s, and uses grub-setup
-to install grub into the boot sector.\n" "$self" "$grubdir";echo
+gettext_printf "%s copies GRUB images into %s.  On some platforms, it
+may also install GRUB into the boot sector.\n" "$self" "$grubdir";echo
 echo
 gettext "Report bugs to <bug-grub@gnu.org>."; echo
 }
-- 
1.8.1.4

